t3/main.py

```diff
@@ -265,11 +288,40 @@ def execute(self):
 
             # SA
             if self.t3['sensitivity'] is not None:
-                sa_success = self.run_sa()
-                if not sa_success:
-                    self.logger.error(f"Could not complete the sensitivity analysis using "
-                                      f"{self.t3['sensitivity']['adapter']}.")
-                    self.trsh_rmg_tol()
+                # determine species to run SA for
+                sa_observables = list()
+                for species in self.rmg['species']:
+                    if species['observable'] or species['SA_observable']:
+                        sa_observables.append(species['label'])
+
+                # use simulate_factory to create the corresponding adapter for SA
+                if 'RMS' in self.t3['sensitivity']['adapter']:
```

Comment:
I thought this would only work if each adapter was uniquely identified by the string used to register it?

Reply:
My understanding is that we could have adapters with the following names:
`RMG`
`RMS batch PT`
`RMS Flow HT`
`Cantera Batch liquid VT`
`Joes Cantera`

Since we should have different adapters for each type of reactor in RMS / Cantera, and we do allow for users to write their own, in which case the adapter name might not have a standard name.

If we have different visions for use cases, let's discuss it.

Reply:
I see we can do that approach. To make sure I understand, you’re suggesting we have one adapter class (file) for each software and reactor type? Ex: `rms_constantV_simulator.py`, `rms_constantTP_simulator.py`, `cantera_constantTP_simulator.py`, `cantera_constantTP_simulator.py`, etc? And then register each of these with the factory was `RMS ConstantV`, `RMS ConstantTP`, etc?

In this case, the only difference between `RMS ConstantV` and `RMS ConstantTP` adapters etc will be one line where we define the domain. Same for `Cantera ConstantV` and `Cantera ConstantTP` etc.  So this will lead to duplicated code and tests that have to be maintained. But I agree that having one file per software and reactor type will make it easier for users to add their own, so I can reorganize the adapters in this way.